diff -ruN a/src/programs/pkcheck.c b/src/programs/pkcheck.c
--- a/src/programs/pkcheck.c	2022-02-02 03:58:54.199404221 +0000
+++ b/src/programs/pkcheck.c	2022-02-02 04:00:23.484557147 +0000
@@ -93,6 +93,11 @@
   allow_user_interaction = FALSE;
   ret = 126;
 
+  if (argc < 1)
+    {
+      exit(126);
+    }
+
   g_type_init ();
 
   details = polkit_details_new ();
diff -ruN a/src/programs/pkexec.c b/src/programs/pkexec.c
--- a/src/programs/pkexec.c	2022-02-02 03:59:01.164494360 +0000
+++ b/src/programs/pkexec.c	2022-02-02 04:02:09.738929657 +0000
@@ -415,6 +415,13 @@
   gchar *opt_user;
   pid_t pid_of_caller;
 
+
+ if (argc<1)
+    {
+      exit(127);
+    }
+
+
   ret = 127;
   authority = NULL;
   subject = NULL;
@@ -520,7 +527,16 @@
           goto out;
         }
       g_free (path);
-      argv[n] = path = s;
+      path = s;
+
+      /* argc<2 and pkexec runs just shell, argv is guaranteed to be null-terminated.
+       * /-less shell shouldn't happen, but let's be defensive and don't write to null-termination
+       */
+      if (argv[n] != NULL)
+      {
+        argv[n] = path;
+      }
+
     }
   if (access (path, F_OK) != 0)
     {
